docs/pages/router/create-pages.mdx

```diff
@@ -112,3 +112,38 @@ export default function Page() {
   return <Text>Blog post: {slug}</Text>;
 }
 ```
+
+## Non-route files
+
+Every file and sub-directory inside the **app** directory is either a `_layout` file, or a route in your app. This means that other files like your components, hooks, utilities, and so on, cannot be placed in the **app** directory because they are neither screens nor layout files.
+
+<FileTree
+  files={[
+    'app/(home)/_layout.tsx',
+    ['app/(home)/profile.tsx', '/profile'],
+    'app/(modal)/_layout.tsx',
+    ['app/(modal)/profile.tsx', '/profile'],
+  ]}
+/>
+
+The example above demonstrates "why", if Expo Router allowed for non-route files, co-location (the technical term for keeping files within the same directory) would lead to an unstructured project. The example is using [shared routes](/router/advanced/shared-routes), and as such there are multiple route files for the `/profile` route. If you have non-route file `ProfileImage.tsx`, you cannot co-locate this file, as you only have one non-route file but two different route directories. To prevent this dilemma Expo Router simply does not allow for non-route file co-location.
+
+### Recommended structure
+
+Expo Router recommends sorting components and hooks by "feature" and routes by "navigation pattern". While the two concepts can overlap, there are enough differences to structure them differently. Additionally, routes will inevitably change. However, components generally stay within the same feature set. Keeping routes separate will greatly simplify your refactors.
```

Comment:
```suggestion
Expo Router recommends sorting components and hooks by "feature" and routes by "navigation pattern". While the two concepts can overlap, there are enough differences to structure them differently. Additionally, routes will inevitably change. However, components generally stay within the same feature set. Keeping routes separate will simplify your refactors.
```

We can remove this phrase since the sentence has enough emphasis about the explanation and providing context on simplification of refactors.